File: ruby/006.rb
#!/usr/bin/env ruby

# http://projecteuler.net/problem=6

def sum_of_squares(max)
  (1..max).collect {|n| n * n }.inject(:+)
end

def square_of_sums(max)
  (1..max).inject(:+) ** 2
end

difference = square_of_sums(100) - sum_of_squares(100)

puts difference

File: erlang/006.erl
#!/usr/bin/env escript

% http://projecteuler.net/problem=6

-mode(compile).

square_of_sums(X) ->
  Numbers = lists:seq(1, X),
  Sum     = lists:sum(Numbers),
  Sum * Sum.

sum_of_squares(X) ->
  Numbers = lists:seq(1, X),
  Squares = lists:map(fun(X) -> X * X end, Numbers),
  lists:sum(Squares).

main(_) ->
  Result = square_of_sums(100) - sum_of_squares(100),
  io:format("~p~n", [Result]).
